plugins/ssrmatching/ssrmatching.ml

```diff
@@ -725,13 +724,13 @@ let rec uniquize = function
       0, uniquize (instances ())
     | NoMatch when (not raise_NoMatch) ->
       if !failed_because_of_TC then
-        errorstrm (source ()++strbrk"matches but type classes inference fails")
+        errorstrm (source env (Evd.from_env env) (* XXX *) ++strbrk"matches but type classes inference fails")
```

Comment:
@gares which `sigma` should I use here?

Reply:
I guess `ise` but I don't think source should be abstrcted, since the term it prints is inside its closure (so it is always the same, no point in printing it under a different context). (Eg, In the first case of `source` the term is printed using `ise`)

Reply:
(source is just a thunk to not compute the error message if we don't need to)